compiler/rustc_mir_transform/src/dead_store_elimination.rs

```diff
@@ -84,7 +86,13 @@ impl<'tcx> MirPass<'tcx> for DeadStoreElimination {
     }
 
     fn run_pass(&self, tcx: TyCtxt<'tcx>, body: &mut Body<'tcx>) {
-        let borrowed = borrowed_locals(body);
-        eliminate(tcx, body, &borrowed);
+        let mut always_live = borrowed_locals(body);
+        if preserve_debug_even_if_never_generated(tcx.sess.opts.debuginfo) {
+            // We model locals that appear in debuginfo as live on exit.
+            // This allows to remove overwritten stores, without completely removing the locals.
+            let debuginfo = debuginfo_locals(body);
+            always_live.union(&debuginfo);
```

Comment:
I don't think the implementation and comment match? `always_live` locals are considered live... well, always, not just on exit, right? The implementation seems to match what was discussed in the other PR though, so probably the comment just needs fixing